library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -1172,6 +1178,9 @@ private void clearRenderedFirstFrame() {
       MediaCodec codec = getCodec();
       // If codec is null then the listener will be instantiated in configureCodec.
       if (codec != null) {
+        if (tunnelingOnFrameRenderedListener != null) {
```

Comment:
Have to do the destroy unconditionally to protect against a queued render that will run and see the new listener.

Reply:
I mean do the destroy whether or not there is a codec.

Reply:
Thanks Bryan, made this fix in commit 43700ea